hazelcast/src/test/java/com/hazelcast/test/TestHazelcastInstanceFactory.java

```diff
@@ -40,64 +39,57 @@
 import static com.hazelcast.instance.TestUtil.terminateInstance;
 import static com.hazelcast.test.HazelcastTestSupport.getAddress;
 import static com.hazelcast.util.Preconditions.checkNotNull;
+import static java.util.Collections.emptyList;
+import static java.util.Collections.unmodifiableCollection;
 
 public class TestHazelcastInstanceFactory {
 
     private static final AtomicInteger PORTS = new AtomicInteger(5000);
 
+    private final ConcurrentMap<Integer, Address> addressMap = new ConcurrentHashMap<Integer, Address>();
     private final boolean mockNetwork = TestEnvironment.isMockNetwork();
     private final AtomicInteger nodeIndex = new AtomicInteger();
 
-    protected TestNodeRegistry registry;
-    protected final ConcurrentMap<Integer, Address> addressMap = new ConcurrentHashMap<Integer, Address>();
     private final int count;
+    private final TestNodeRegistry registry;
+
+    public TestHazelcastInstanceFactory() {
+        this(0);
+    }
 
     public TestHazelcastInstanceFactory(int count) {
         this.count = count;
-        if (mockNetwork) {
-            List<Address> addresses = createAddresses(PORTS, count);
-            initFactory(addresses);
-        }
+        this.registry = getRegistry(createAddresses(PORTS, count));
     }
 
-    public TestHazelcastInstanceFactory() {
-        this.count = 0;
-        this.registry = new TestNodeRegistry(Collections.unmodifiableCollection(addressMap.values()));
+    public TestHazelcastInstanceFactory(String... addresses) {
+        this(-1, addresses);
     }
 
     public TestHazelcastInstanceFactory(int initialPort, String... addresses) {
         this.count = addresses.length;
-        if (mockNetwork) {
-            List<Address> addressList = createAddresses(initialPort, PORTS, addresses);
-            initFactory(addressList);
-        }
-    }
-
-    public TestHazelcastInstanceFactory(String... addresses) {
-        this.count = addresses.length;
-        if (mockNetwork) {
-            List<Address> addressesList = createAddresses(-1, PORTS, addresses);
-            initFactory(addressesList);
-        }
+        this.registry = getRegistry(createAddresses(initialPort, PORTS, addresses));
     }
 
     public TestHazelcastInstanceFactory(Collection<Address> addresses) {
         this.count = addresses.size();
-        if (mockNetwork) {
-            initFactory(addresses);
-        }
+        this.registry = getRegistry(addresses);
     }
 
-    private void initFactory(Collection<Address> addresses) {
+    private TestNodeRegistry getRegistry(Collection<Address> addresses) {
```

Comment:
If you create something, it is best to call it new/create. This way you make it clear a new object instance is being created instead of some available object being returned.